File: app/iosApp/Sources/Features/System/SystemStatusScreen.swift
import SwiftUI

struct SystemStatusScreen: View {
  let model: SystemStatusViewModel

  var body: some View {
    VStack(spacing: AppSpacing.md) {
      Text("System status").font(.title2).foregroundStyle(AppColors.onSurface)
      status
      AppButton(label: "Refresh") { Task { await model.refresh() } }
    }
    .padding(AppSpacing.lg)
    .frame(maxWidth: .infinity, maxHeight: .infinity)
    .background(AppColors.background)
    .task { await model.refresh() }
  }

  @ViewBuilder private var status: some View {
    switch model.state {
    case .loading:
      ProgressView().accessibilityLabel("Loading system status")
    case .content(let databaseReady):
      Text(databaseReady ? "Database ready" : "Database not ready")
        .accessibilityIdentifier("database-status")
        .foregroundStyle(AppColors.onSurface)
    case .failure:
      Text("Unable to load system status.")
        .accessibilityIdentifier("system-error")
        .foregroundStyle(AppColors.onSurface)
    }
  }
}

File: app/androidApp/src/main/kotlin/com/example/template/feature/system/SystemStatusScreen.kt
package com.example.template.feature.system

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.safeDrawingPadding
import androidx.compose.material3.CircularProgressIndicator
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.semantics.contentDescription
import androidx.compose.ui.semantics.semantics
import androidx.compose.ui.semantics.testTag
import com.example.template.core.designsystem.AppButton
import com.example.template.core.designsystem.AppColors
import com.example.template.core.designsystem.AppSpacing

@Composable
fun SystemStatusScreen(state: SystemStatusScreenState, onRefresh: () -> Unit) {
    Surface(color = AppColors.background, contentColor = AppColors.onSurface) {
        Column(
            modifier = Modifier.fillMaxSize().safeDrawingPadding().padding(AppSpacing.lg),
            verticalArrangement = Arrangement.spacedBy(AppSpacing.md, Alignment.CenterVertically),
            horizontalAlignment = Alignment.CenterHorizontally,
        ) {
            Text(
                "System status",
                style = MaterialTheme.typography.titleLarge,
                color = AppColors.onSurface,
            )
            when (state) {
                SystemStatusScreenState.Loading ->
                    CircularProgressIndicator(
                        Modifier.semantics {
                            testTag = "system-loading"
                            contentDescription = "Loading system status"
                        }
                    )
                is SystemStatusScreenState.Content ->
                    Text(
                        text = if (state.databaseReady) "Database ready" else "Database not ready",
                        modifier = Modifier.semantics { testTag = "database-status" },
                    )
                is SystemStatusScreenState.Error ->
                    Text(state.message, modifier = Modifier.semantics { testTag = "system-error" })
            }
            AppButton(
                "Refresh",
                onRefresh,
                Modifier.fillMaxWidth().semantics { testTag = "refresh-button" },
            )
        }
    }
}
